scalafmt-dynamic/src/test/scala/tests/ScalafmtVersionSuite.scala

```diff
@@ -0,0 +1,42 @@
+package tests
+
+import org.scalatest.FunSuite
+import org.scalafmt.dynamic.ScalafmtVersion
+import org.scalafmt.dynamic.ScalafmtVersion.InvalidVersionException
+
+class ScalafmtVersionSuite extends FunSuite {
+  test("parse valid versions") {
+    assert(ScalafmtVersion.parse("2.0.0") == Right(ScalafmtVersion(2, 0, 0, 0)))
+    assert(ScalafmtVersion.parse("0.1.3") == Right(ScalafmtVersion(0, 1, 3, 0)))
+    assert(
+      ScalafmtVersion.parse("2.0.0-RC4") == Right(ScalafmtVersion(2, 0, 0, 4))
+    )
+    assert(ScalafmtVersion.parse("2.1.1") == Right(ScalafmtVersion(2, 1, 1, 0)))
+  }
+
+  test("fail on invalid versions") {
+    assert(ScalafmtVersion.parse("2.0") == Left(InvalidVersionException("2.0")))
+    assert(
+      ScalafmtVersion.parse("v2.0.0") == Left(InvalidVersionException("v2.0.0"))
+    )
+    assert(ScalafmtVersion.parse("avs") == Left(InvalidVersionException("avs")))
+    assert(
+      ScalafmtVersion
+        .parse("1.2.3-M14") == Left(InvalidVersionException("1.2.3-M14"))
+    )
```

Comment:
Could you add some more invalid examples? like

- `2.-1.0` (maybe we should check all the fields are positive or 0)
- `2.1.0.` 
- `2.1.0.1`
- `2.1.0-rc1` (small letter)